static/app/views/dashboardsV2/widgetBuilder/groupBy.tsx

```diff
@@ -0,0 +1,137 @@
+      columns.length === 0
+        ? [{...EMPTY_FIELD}, {...EMPTY_FIELD}]
+        : [...columns, {...EMPTY_FIELD}];
+    onChange(newColumns);
+  }
+
+  function handleSelect(value: QueryFieldValue, index?: number) {
+    const newColumns = [...columns];
+    if (columns.length === 0) {
+      newColumns.push(value);
+    } else if (defined(index)) {
+      newColumns[index] = value;
+    }
+    onChange(newColumns);
+  }
+
+  function handleRemove(index: number) {
+    const newColumns = [...columns];
+    newColumns.splice(index, 1);
+    onChange(newColumns);
+  }
+
+  if (columns.length === 0) {
+    return (
+      <React.Fragment>
+        <StyledField inline={false} flexibleControlStateSize stacked>
+          <QueryFieldWrapper>
+            <QueryField
+              placeholder={t('Select group')}
+              fieldValue={EMPTY_FIELD}
+              fieldOptions={fieldOptions}
+              onChange={value => handleSelect(value, 0)}
+              filterPrimaryOptions={filterPrimaryOptions}
+            />
+          </QueryFieldWrapper>
+        </StyledField>
+
+        <AddGroupButton size="small" icon={<IconAdd isCircled />} onClick={handleAdd}>
+          {t('Add Group')}
+        </AddGroupButton>
+      </React.Fragment>
+    );
+  }
+
+  const hasOnlySingleColumnWithValue =
+    columns.length === 1 &&
+    columns[0].kind === FieldValueKind.FIELD &&
+    columns[0]?.field !== '';
+
+  const canDelete = columns.length > 1 || hasOnlySingleColumnWithValue;
+
+  return (
+    <React.Fragment>
+      <StyledField inline={false} flexibleControlStateSize stacked>
+        {columns?.map((column, index) => (
+          <QueryFieldWrapper key={`groupby-${index}`}>
+            <QueryField
+              placeholder={t('Select group')}
+              fieldValue={column}
+              fieldOptions={fieldOptions}
+              onChange={value => handleSelect(value, index)}
+              filterPrimaryOptions={filterPrimaryOptions}
+            />
+            {canDelete && (
+              <Button
+                size="zero"
+                borderless
+                onClick={() => handleRemove(index)}
+                icon={<IconDelete />}
+                title={t('Remove group')}
+                aria-label={t('Remove group')}
+              />
+            )}
+          </QueryFieldWrapper>
+        ))}
+      </StyledField>
+
+      {columns.length < GROUP_BY_LIMIT && (
+        <AddGroupButton size="small" icon={<IconAdd isCircled />} onClick={handleAdd}>
+          {t('Add Group')}
+        </AddGroupButton>
+      )}
+    </React.Fragment>
+  );
+}
+
+const StyledField = styled(Field)`
+  padding-bottom: ${space(1)};
+`;
+
+const QueryFieldWrapper = styled('div')`
+  display: flex;
+  align-items: center;
+  justify-content: space-between;
+
+  :not(:last-child) {
+    margin-bottom: ${space(1)};
+  }
+
+  > * + * {
+    margin-left: ${space(1)};
+  }
+`;
+
+const AddGroupButton = styled(Button)`
+  width: min-content;
```

Comment:
what happens if we don't define this style?

Reply:
<img width="953" alt="Screen Shot 2022-03-11 at 10 10 56 AM" src="https://user-images.githubusercontent.com/22846452/157895269-bcac4a68-de4b-4fb0-be57-1ecd149d65db.png">

Without this, the button gets long and fills the grid row. I know we've been typically using a styled ButtonBar and setting `justify-content: flex-start` and then they're the right size, but since this is just a single button I was wondering if that was unnecessary here